pkg/opensearch/client/search_request_test.go

```diff
@@ -41,12 +41,68 @@ func TestSearchRequest(t *testing.T) {
 			})
 		})
 
+		t.Run("When adding nested bool query filters, When building search request", func(t *testing.T) {
+			b := NewSearchRequestBuilder(OpenSearch, version, tsdb.Interval{Value: 15 * time.Second, Text: "15s"})
+			b.Size(200)
+			b.Sort("desc", timeField, "boolean")
+			filters := b.Query().Bool().Filter()
+			filters.AddFilterQuery(Query{
+				&BoolQuery{
+					ShouldFilters: []Filter{
+						Query{
+							&BoolQuery{
+								Filters: []Filter{
+									Query{&BoolQuery{MustNotFilters: []Filter{TermsFilter{
+										Key:    "parentSpanId",
+										Values: []string{""},
+									}}}},
+								},
+							},
+						},
+					},
+				},
+			})
+			sr, err := b.Build()
+			assert.NoError(t, err)
+
+			t.Run("Should have correct size", func(t *testing.T) {
+				assert.Equal(t, 200, sr.Size)
+			})
+			t.Run("Should have must not filter inside a should filter", func(t *testing.T) {
+				boolQuery, ok := sr.Query.Bool.Filters[0].(Query)
+				assert.True(t, ok)
+				mustNotFilters := boolQuery.Bool.ShouldFilters[0].(Query).Bool.Filters[0].(Query).Bool.MustNotFilters[0].(TermsFilter)
+				assert.Equal(t, "parentSpanId", mustNotFilters.Key)
+				assert.Equal(t, []string{""}, mustNotFilters.Values)
+			})
+			t.Run("Should have a terms filter inside a must not filter", func(t *testing.T) {
+				boolQuery, ok := sr.Query.Bool.Filters[0].(Query)
+				assert.True(t, ok)
+				mustNotFilters := boolQuery.Bool.ShouldFilters[0].(Query).Bool.Filters[0].(Query).Bool.MustNotFilters[0].(TermsFilter)
+				assert.Equal(t, "parentSpanId", mustNotFilters.Key)
+				assert.Equal(t, []string{""}, mustNotFilters.Values)
+			})
+			t.Run("When marshal to JSON should generate correct json", func(t *testing.T) {
+				body, err := json.Marshal(sr)
+				assert.NoError(t, err)
+				json, err := simplejson.NewJson(body)
+				assert.NoError(t, err)
+
+				parentSpanId, err := json.GetPath("query", "bool", "filter", "bool", "should", "bool", "filter", "bool", "must_not", "term", "parentSpanId", "value").String()
+				assert.NoError(t, err)
+
+				assert.Equal(t, "", parentSpanId)
+
```

Comment:
style nit: group function calls in the same code block as their asserts
```suggestion
				assert.NoError(t, err)
				assert.Equal(t, "", parentSpanId)
```